server/src/main/java/org/eclipse/openvsx/ExtensionService.java

```diff
@@ -52,22 +58,36 @@ public class ExtensionService {
     @Value("${ovsx.publishing.require-license:false}")
     boolean requireLicense;
 
+    @Transactional
+    public ExtensionVersion mirrorVersion(Path extensionFile, PersonalAccessToken token, String binaryName, String timestamp) {
+        var downlaod = doPublish(extensionFile, token, TimeUtil.fromUTCString(timestamp));
+        publishHandler.mirror(downlaod, extensionFile);
+        downlaod.setName(binaryName);
+        return downlaod.getExtension();
+    }
+
     public ExtensionVersion publishVersion(InputStream content, PersonalAccessToken token) {
-        FileResource download;
-        ExtensionVersion extVersion;
+        // TODO delete extension file after publishing
```

Comment:
We hit out of disk situation on `/tmp` folder while mirroring, because we generate to many vsix files before processing. So in the mirror mode we added a logic everywhere to check that files are deleted after mirroring.

We put here TODO because it is probably can happen in the usual mode as well if a pod running for very long time. If you don't think so then let's remove TODO.

Reply:
Files in the `/tmp` directory get cleaned up by the system. However this is every 10 to 30 days or on every restart. You can use `systemd-tmpfiles` to configure a different delete interval.
It's a good practice to delete files after the server no longer needs them. #557 will most likely increase the amount of temp files generated by the server quite a bit. I've added this TODO to #557, you can remove it in the code.

Reply:
cc @iQQBot do you know we can do something like that?